PublicFolders/ValidatePFDumpster.ps1

```diff
@@ -0,0 +1,416 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$ts = Get-Date -Format yyyyMMdd_HHmmss
+$ExportPath = "$env:USERPROFILE\Desktop\ValidatePFDumpster\ValidatePFDumpster_$ts"
+mkdir $ExportPath -Force | Out-Null
+$Script:Conditionsfailed = 0
+Function log {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$CurrentStatus,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Function,
+
+        [Parameter(Mandatory = $true)]
+        [string]$CurrentDescription
+
+    )
+    [PSCustomObject]@{
+        Function    = $Function
+        Description = $CurrentDescription
+        Status      = $CurrentStatus
+    } | Export-Csv $ExportPath\ValidatePFDumpsterCheckslogging.csv -NoTypeInformation -Append
+}
+Function WritetoScreenANDlog {
+    param(
+        [Parameter(Mandatory = $true)]
+        [String]$Issue,
+
+        [Parameter(Mandatory = $true)]
+        [String]$Fix
+    )
+    Write-Host $Issue -ForegroundColor Red
+    $Issue | Out-File $ExportPath\ValidatePFDumspterREPORT.txt -Append
+    Write-Host $Fix -ForegroundColor Green
+    $Fix+"`n" | Out-File $ExportPath\ValidatePFDumspterREPORT.txt -Append
+    Write-Host
+}
+
+Function Connect2EXO {
+    try {
+        #Requires -Modules @{ModuleName="ExchangeOnlineManagement"; ModuleVersion="2.0.0" }
```

Comment:
Added #Requires at the start of the script but I'll leave version to 2.0.0 as it's just 2 months release so I don't want to keep that module a blocker for admins in case they are not able to install/update to that recent module.